managedplugin/download.go

```diff
@@ -0,0 +1,222 @@
+package managedplugin
+
+import (
+	"archive/zip"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/avast/retry-go/v4"
+	"github.com/schollz/progressbar/v3"
+)
+
+const (
+	DefaultDownloadDir = ".cq"
+	RetryAttempts      = 5
+	RetryWaitTime      = 1 * time.Second
+)
+
+// getURLLocation return the URL of the plugin
+// this does a few HEAD requests because we had a few breaking changes to where
+// we store the plugins on GitHub
+// TODO: we can improve this by just embedding all plugins and last version that exist in different places then
+// the latest
+func getURLLocation(ctx context.Context, org string, name string, version string) (string, error) {
+	urls := []string{
+		fmt.Sprintf("https://github.com/%s/cq-plugin-%s/releases/download/%s/cq-plugin-%s_%s_%s.zip", org, name, version, name, runtime.GOOS, runtime.GOARCH),
+		fmt.Sprintf("https://github.com/%s/cq-source-%s/releases/download/%s/cq-source-%s_%s_%s.zip", org, name, version, name, runtime.GOOS, runtime.GOARCH),
+		fmt.Sprintf("https://github.com/%s/cq-destination-%s/releases/download/%s/cq-destination-%s_%s_%s.zip", org, name, version, name, runtime.GOOS, runtime.GOARCH),
+	}
+	if org == "cloudquery" {
+		urls = append(urls, fmt.Sprintf("https://github.com/cloudquery/cloudquery/releases/download/plugins-plugin-%s-%s/%s_%s_%s.zip", name, version, name, runtime.GOOS, runtime.GOARCH))
```

Comment:
```suggestion
		urls = append(urls, fmt.Sprintf("https://github.com/cloudquery/cloudquery/releases/download/plugins-%s-%s/%s_%s_%s.zip", name, version, name, runtime.GOOS, runtime.GOARCH))
```

I think if we moved to this structure (which IMO we shouldn't in this migration, but maybe later), we'd want all plugins to live directly under `plugins/` without the additional `plugins/plugin` layer

Reply:
agree. I kept the structure the same for now and also removed the additional `HEAD` as we don't have any plugins there for now.